docs/developer-guide.md

```diff
@@ -48,3 +48,67 @@ const showFlaggedFeature = useShowGrowthbookFeature('flagged-feature'); // Set t
+
+## Database Testing
+
+### pgTap + sqitch tips
+
+#### Run a single test
+
+You can run a single test by calling `pg_prove` directly from the command line like so:
+`pg_prove -v -d \<test_db_name\> \<path-to-test-file\>/file_to_test.sql`
+flags:
+
+- `-v`runs pg_prove in verbose. This allows you to add select statements in the test file & they will show up in the output (helpful for debugging)
+- `-d` sets the target database (must be right before your test database)
+
+The path to the test file can be a glob pattern. So if you want to run all tests in a specific folder, or don't feel like writing our the whole path,
+you could write the command like this (assuming our tests are found at `test/unit/tables/file_test.sql`:
+
+Run all table tests:
+`pg_prove -v -d \<test_db_name\> test/unit/tables/*_test.sql`
+Run a specific test anywhere in the 'unit' directory:
+`pg_prove -v -d \<test_db_name\> test/unit/**/specific_file_test.sql`
+
+#### Idempotence (create or replace function...)
+
+Our make target `make db_unit_tests` drops the database, recreates it and runs all sqitch migrations.
+This is annoying if you just want to test the last thing you deployed (which is usually the case).
+With idempotent migrations (like functions) you can make changes to your deploy file and copy paste the `create function..` statement directly into a terminal connected to your test db to save some time.
+
+#### Revert & Deploy with a target
+
+If the change you're testing is not idempotent (like creating or altering a table), then you'll want to revert/deploy just that change in your test database
+as you make changes to avoid having to redeploy all changes just to get the change you've made to apply.
+Sqitch by default is probably pointed at the local database that you use for running the app locally.
+Here we just want to target the test database and only the change you're testing.
+
+To revert to a specific change in your sqitch plan (on your test_db):
+`sqitch revert \<name-of-change\> \<your-test-db\>`
+Same thing to deploy, if you want to to deploy to the end of your migrations:
+`sqitch deploy \<your-test-db\>`
+Or if you'd like to deploy up to a specific migration:
+`sqitch deploy \<name-of-change\> \<your-test-db\>`
+
+#### Debugging in the test file
+
+Using the -v option is helpful to be able to add some debugging to the output in your terminal when the test runs.
+It will also run the output of any functions your test writes, so the output can get a little hard to read.
+Adding some `select` statements to debug will help to debug a flaky or tricky test, but wraping these statments in "breakers" can make them much more readable.
+You can write some wrappers to break up the output and give a title to what you are debugging :
+
+select '---------DEBUG 1---------';
+select 'Value before commit:';
+select \<debug statement\>
+select '--------END DEBUG 1--------';
+select '---------DEBUG 2---------;
+...
+...
```

Comment:
They all end up on the same line otherwise

```suggestion
    ````
    select '---------DEBUG 1---------';
    select 'Value before commit:';
    select \<debug statement\>
    select '--------END DEBUG 1--------';
    select '---------DEBUG 2---------;
    ...
    ...
    ````
```